source/SkiaSharp.Views/SkiaSharp.Views.WinUI/SKXamlCanvas.cs

```diff
@@ -101,8 +101,13 @@ private static void OnVisibilityChanged(DependencyObject d, DependencyPropertyCh
 		private void OnXamlRootChanged(XamlRoot xamlRoot = null, XamlRootChangedEventArgs e = null)
 		{
 			var root = xamlRoot ?? XamlRoot;
-			Dpi = root?.RasterizationScale ?? 1.0;
-			Invalidate();
+			var newDpi = root?.RasterizationScale ?? 1.0;
+			if (newDpi != Dpi)
```

Comment:
This check here prevents hundreds of invalidations. The `XamlRootChanged` event will fire when _any_ of the properties of the XAML root changes. This does not seem to fire just when the current view gets a new XAML root as the name implies.